content/systems/filecoin_files/data_transfer/_index.md

```diff
@@ -23,25 +27,29 @@ but their code belongs in the Markets system.
+- **Requestor**: The party that initiates the data transfer request (whether Push or Pull) - normally the client, at least as currently implemented in Filecoin, to overcome NAT-traversal problems.
+- **Responder**: The party that receives the data transfer request - normally the storage provider.
+- **Data Transfer Voucher or Token**: A wrapper around storage- or retrieval-related data that can identify and validate the transfer request to the other party.
+- **Request Validator**: The data transfer module only initiates a transfer when the responder can validate that the request is tied directly to either an existing storage or retrieval deal. Validation is not performed by the data transfer module itself. Instead, a request validator inspects the data transfer voucher to determine whether to respond to the request or disregard the request.
+- **Transporter**:  Once a request is negotiated and validated, the actual transfer is managed by a transporter on both sides. The transporter is part of the data transfer module but is isolated from the negotiation process. It has access to an underlying verifiable transport protocol and uses it to send data and track progress.
 - **Subscriber**: An external component that monitors progress of a data transfer by subscribing to data transfer events, such as progress or completion.
-- **GraphSync**: The default underlying transfer protocol used by the Scheduler. The full graphsync specification can be found [here](https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md)
+- **GraphSync**: The default underlying transport protocol used by the Transporter. The full graphsync specification can be found [here](https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md)
 
 ## Request Phases
 
 There are two basic phases to any data transfer:
 
-1. Negotiation - the requestor and responder agree to the transfer by validating with the data transfer voucher
-2. Transfer - Once the negotiation phase is complete, the data is actually transferred. The default protocol used to do the transfer is Graphsync.
+1. Negotiation: the requestor and responder agree to the transfer by validating it with the data transfer voucher.
+2. Transfer: once the negotiation phase is complete, the data is actually transferred. The default protocol used to do the transfer is Graphsync.
 
 Note that the Negotiation and Transfer stages can occur in separate round trips,
-or potentially the same round trip, where the requesting party implicitly agrees by sending the request, and the responding party can agree and immediately send or receive data.
+or potentially the same round trip, where the requesting party implicitly agrees by sending the request, and the responding party can agree and immediately send or receive data. Whether the process is taking place in a single or multiple round-trips depends in part on whether the request is a push request (storage deal) or a pull request (retrieval deal).
+
```

Comment:
Would add:

```suggestion
In the case of GraphSync as transport mechanism, data transfer requests can piggy back as an extension to the GraphSync protocol using [GraphSync's built-in extensibility](https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md#extensions). So, only a single round trip is required for Pull Requests. However, because Graphsync is a request/response protocol with no direct support for `push` type requests, in the Push case, negotiation happens in a seperate request over data transfer's own libp2p protocol `/fil/datatransfer/1.0.0`. Other future transport mechinisms might handle both Push and Pull, either, or neither as a single round trip.